activerecord/lib/active_record/relation/from_clause.rb

```diff
@@ -25,7 +25,7 @@ def empty?
       end
 
       def self.empty
-        new(nil, nil)
+        @empty ||= new(nil, nil).freeze
```

Comment:
This change also doesn't enforce it, as it only affects the object returned by `empty`. At the end of the day I consider this change to be unnecessary, as would a change that freezes `self ` in `initialize`. Objects have an API for a reason.